File: 2022-07-25/programmers_기능개발_말감.py
def solution(progresses, speeds):
    answer = []
    while progresses :
        if progresses[0] >= 100 :
            progresses.pop(0)
            speeds.pop(0)
            answer.append(1)
            while progresses and progresses[0] >= 100 :
                progresses.pop(0)
                speeds.pop(0)
                answer[-1] += 1
            continue
        for i in range(len(progresses)) :
            progresses[i] += speeds[i]
    return answer

File: 9월/2022-09-05/programmers_기능개발_말감.js
function solution(progresses, speeds) {
  const answer = [];
  let day = 0;
  progresses.map((progress, index) => {
    if (speeds[index] * day + progress >= 100) {
      answer[answer.length - 1] += 1;
    } else {
      day = Math.ceil((100 - progress) / speeds[index]);
      answer.push(1);
    }
  });
  return answer;
}
